guacamole/src/main/webapp/app/client/types/ManagedClient.js

```diff
@@ -577,13 +582,28 @@ angular.module('client').factory('ManagedClient', ['$rootScope', '$injector',
                 managedClient.filesystems.push(ManagedFilesystem.getInstance(object, name));
             });
         };
+        
+        // Handle any received prompts
+        client.onrequired = function onrequired(parameters) {
+            
+            guacPrompt.getUserInput(parameters.reduce((a,b)=> (a[b]='',a),{}))
+                    .then(function gotUserInput(data) {
+                        for (var parameter in data) {
+                            var stream = client.createArgumentValueStream("text/plain", parameter);
+                            var writer = new Guacamole.StringWriter(stream);
+                            writer.sendText(data[parameter]);
+                            writer.sendEnd();
+                        }
+
+                }, function errorUserInput() {
+                    $log.error('Error gathering user input.');
+                    client.disconnect();
+                });
```

Comment:
@necouchman In my initial testing I am getting a popup that looks like this:
![image](https://user-images.githubusercontent.com/64200338/80146652-b28c7600-85a1-11ea-9c22-924f458ea34d.png)

Looks like we may be missing the client.forms for the content field in guacPrompt.html, where was that intended to be populated?

Reply:
Upon further testing this morning, this use case happens when the user clicks on a CONNECTION_GROUP to initiate a connection rather than a CONNECTION.

Reply:
Ah, yes - okay, good to know.  I'll have to figure out how to pull the forms for the underlying connection...

Reply:
I was having difficulty with that as well.  I couldn't find an easy way to identify the childrenConnections protocol.  The only way I saw was to just pull the protocol from the first child connection, which breaks pretty easily (especially since I think this is meant to be able to support connection-groups with sub-connection-groups.

Is there a reason the Connection-group is passed around as the selected object and then later assigned a specific connection?  Would there be a way get the 'assigned' connection first and then pass around that clientIdentifier?

Reply:
> Is there a reason the Connection-group is passed around as the selected object and then later assigned a specific connection? Would there be a way get the 'assigned' connection first and then pass around that clientIdentifier?

I think at least one of the reasons is that the user accessing the connection need not have any knowledge of or even access to the underlying connection when using a load balancing connection group.  That is, you can assign users access to the top-level connection group and not to anything else, and they'll still be able to connect.  The current behavior insures that this works, and I think what you're suggesting would require that the user accessing the connection group also be able to see/read the underlying connection.  But, I'm not sure if there's any good way around this.

@mike-jumper Could use your expertise, here :-).

Reply:
Will there ever be a case where a Connection-group has more than one protocol?

Reply:
Well, it is certainly possible today, yes, and I'm not sure that we should go the route of forcing things into that mold (one protocol per Connection Group) just yet.  While I do not have any scenarios in my current usage of Guacamole that require that, there may well be others that do.